src/main/java/com/ericsson/ei/queryservice/ProcessQueryParams.java

```diff
@@ -63,26 +67,70 @@
      * @return JSONArray
      * @throws IOException
      */
-    public JSONArray filterFormParam(JSONObject criteriaObj, JSONObject optionsObj) {
+    public JSONArray filterFormParam(JSONObject criteriaObj, JSONObject optionsObj, String filter) {
         JSONArray resultAggregatedObject;
         String criteria = editObjectNameInQueryParam(criteriaObj);
         
         
         if (optionsObj == null || optionsObj.toString().equals("{}")) {
             resultAggregatedObject = processAggregatedObject.processQueryAggregatedObject(criteria, databaseName, aggregationCollectionName);
         } else {
-        	String options = editObjectNameInQueryParam(optionsObj); 
+            String options = editObjectNameInQueryParam(optionsObj);
             LOGGER.debug("The options is : " + options);
-            String result = "{ \"$and\" : [ " + criteria + "," + options + " ] }";
-            resultAggregatedObject = processAggregatedObject.processQueryAggregatedObject(result, databaseName, aggregationCollectionName);
+            String request = "{ \"$and\" : [ " + criteria + "," + options + " ] }";
+            resultAggregatedObject = processAggregatedObject.processQueryAggregatedObject(request, databaseName, aggregationCollectionName);
         }
-        LOGGER.debug("resultAggregatedObject : " + resultAggregatedObject.toString());
+        resultAggregatedObject = checkFilterCondition(filter, resultAggregatedObject);
+
         return resultAggregatedObject;
     }
 
     /**
-     * This method takes the parameters from the REST GET request query. If the
-     * Aggregated Object matches the condition, then it is returned.
+     * This method checks if filter condition exists.
+     *
+     * @param filterKey, resultAggregatedObjectArray
+     * @return JSONArray
+     * @throws IOException
+     */
+    private JSONArray checkFilterCondition(String filter, JSONArray resultAggregatedObject) {
+        if (filter == null || filter.equals("")) {
+            LOGGER.debug("resultAggregatedObject : " + resultAggregatedObject.toString());
```

Comment:
this message should be something like no filter conditions provided.

The current log does not do anything helpful.

Reply:
Done